? .colordiff.pl.swp
? colordiff-colors.patch
? colordiff_1.0.5-1.diff.gz
? cvs.diffy
? foo1
? foo2
? plain.diffy
Index: .cvsignore
===================================================================
RCS file: /cvsroot/colordiff/colordiff/src/Attic/.cvsignore,v
retrieving revision 1.1.2.2
diff --side-by-side -r1.1.2.2 .cvsignore
colordiff.1							colordiff.1
							      >	cdiff.1
README								README
Index: Makefile
===================================================================
RCS file: /cvsroot/colordiff/colordiff/src/Makefile,v
retrieving revision 1.2.2.12
diff --side-by-side -r1.2.2.12 Makefile
# $Id: Makefile,v 1.2.2.12 2005/03/28 21:07:03 daveewart Exp  <
							      <
INSTALL_DIR=/usr/local/bin					INSTALL_DIR=/usr/local/bin
MAN_DIR=/usr/local/man/man1					MAN_DIR=/usr/local/man/man1
ETC_DIR=/etc							ETC_DIR=/etc
VERSION=1.0.5						      |	VERSION=1.0.6
DIST_FILES=COPYING INSTALL Makefile README \			DIST_FILES=COPYING INSTALL Makefile README \
	colordiff.pl colordiffrc colordiffrc-lightbg cdiff.sh |		colordiff.pl colordiffrc colordiffrc-lightbg cdiff.sh
							      >		colordiff.xml cdiff.xml
TMPDIR=colordiff-${VERSION}					TMPDIR=colordiff-${VERSION}
TARBALL=${TMPDIR}.tar.gz					TARBALL=${TMPDIR}.tar.gz


doc: colordiff.xml					      |	doc: colordiff.xml cdiff.xml
	xmlto -vv man colordiff.xml					xmlto -vv man colordiff.xml
							      >		xmlto -vv man cdiff.xml
	xmlto -vv txt colordiff.xml					xmlto -vv txt colordiff.xml
	xmlto -vv html-nochunks colordiff.xml				xmlto -vv html-nochunks colordiff.xml
	mv colordiff.txt README						mv colordiff.txt README
	perl -p -i -e 's#<head>#<head><link rel=\"stylesheet\		perl -p -i -e 's#<head>#<head><link rel=\"stylesheet\
	perl -p -i -e 's#</body>#</div></body>#' colordiff.ht		perl -p -i -e 's#</body>#</div></body>#' colordiff.ht
	perl -p -i -e 's#<div class=\"refentry\"#<div id=\"co		perl -p -i -e 's#<div class=\"refentry\"#<div id=\"co
	mv colordiff.html ../htdocs					mv colordiff.html ../htdocs

etc:								etc:
	sed -e "s%/etc%$(ETC_DIR)%g" colordiff.pl > colordiff		sed -e "s%/etc%$(ETC_DIR)%g" colordiff.pl > colordiff

install: etc							install: etc
	install -D colordiff.pl.for.install ${INSTALL_DIR}/co		install -D colordiff.pl.for.install ${INSTALL_DIR}/co
	if [ ! -f ${INSTALL_DIR}/cdiff ] ; then \			if [ ! -f ${INSTALL_DIR}/cdiff ] ; then \
	  install cdiff.sh ${INSTALL_DIR}/cdiff; \			  install cdiff.sh ${INSTALL_DIR}/cdiff; \
	fi								fi
	install -D colordiff.1 ${MAN_DIR}/colordiff.1			install -D colordiff.1 ${MAN_DIR}/colordiff.1
	if [ -f ${ETC_DIR}/colordiffrc ]; then \			if [ -f ${ETC_DIR}/colordiffrc ]; then \
	  mv -f ${ETC_DIR}/colordiffrc ${ETC_DIR}/colordiffrc		  mv -f ${ETC_DIR}/colordiffrc ${ETC_DIR}/colordiffrc
	fi								fi
	cp colordiffrc ${ETC_DIR}/colordiffrc				cp colordiffrc ${ETC_DIR}/colordiffrc
	chown root.root ${ETC_DIR}/colordiffrc				chown root.root ${ETC_DIR}/colordiffrc
	chmod 644 ${ETC_DIR}/colordiffrc				chmod 644 ${ETC_DIR}/colordiffrc
	rm -f colordiff.pl.for.install					rm -f colordiff.pl.for.install

dist:								dist:
	mkdir ${TMPDIR}							mkdir ${TMPDIR}
	cp -p ${DIST_FILES} ${TMPDIR}					cp -p ${DIST_FILES} ${TMPDIR}
	tar -zcvf ${TARBALL} ${TMPDIR}					tar -zcvf ${TARBALL} ${TMPDIR}
	rm -fR ${TMPDIR}						rm -fR ${TMPDIR}

clean:								clean:
	rm -f README colordiff.1 colordiff.html		      |		rm -f README colordiff.1 colordiff.html cdiff.1
Index: colordiff.pl
===================================================================
RCS file: /cvsroot/colordiff/colordiff/src/colordiff.pl,v
retrieving revision 1.4.2.12
diff --side-by-side -r1.4.2.12 colordiff.pl
#!/usr/bin/perl -w						#!/usr/bin/perl -w
# 							      <
# $Id: colordiff.pl,v 1.4.2.12 2005/03/28 20:11:34 daveewart  <

#############################################################	#############################################################
#                                                            	#                                                            
# ColorDiff - a wrapper/replacment for 'diff' producing      	# ColorDiff - a wrapper/replacment for 'diff' producing      
#             colourful output                               	#             colourful output                               
#                                                            	#                                                            
# Copyright (C)2002-2004 Dave Ewart (davee@sungate.co.uk)     |	# Copyright (C)2002-2006 Dave Ewart (davee@sungate.co.uk)    
#                                                            	#                                                            
#############################################################	#############################################################
#                                                            	#                                                            
# This program is free software; you can redistribute it and/	# This program is free software; you can redistribute it and/
# it under the terms of the GNU General Public License as pub	# it under the terms of the GNU General Public License as pub
# the Free Software Foundation; either version 2 of the Licen	# the Free Software Foundation; either version 2 of the Licen
# (at your option) any later version.                        	# (at your option) any later version.                        
#                                                            	#                                                            
# This program is distributed in the hope that it will be use	# This program is distributed in the hope that it will be use
# but WITHOUT ANY WARRANTY; without even the implied warranty	# but WITHOUT ANY WARRANTY; without even the implied warranty
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See t	# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See t
# GNU General Public License for more details.               	# GNU General Public License for more details.               
#                                                            	#                                                            
# You should have received a copy of the GNU General Public L <
# along with this program; if not, write to the Free Software <
# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.   <
#                                                             <
#############################################################	#############################################################

use strict;							use strict;
use Getopt::Long qw(:config pass_through);			use Getopt::Long qw(:config pass_through);
use IPC::Open2;							use IPC::Open2;

my $app_name     = 'colordiff';					my $app_name     = 'colordiff';
my $version      = '1.0.5';				      |	my $version      = '1.0.6';
my $author       = 'Dave Ewart';				my $author       = 'Dave Ewart';
my $author_email = 'davee@sungate.co.uk';			my $author_email = 'davee@sungate.co.uk';
my $app_www      = 'http://colordiff.sourceforge.net/';		my $app_www      = 'http://colordiff.sourceforge.net/';
my $copyright    = '(C)2002-2005';			      |	my $copyright    = '(C)2002-2006';
my $show_banner  = 1;						my $show_banner  = 1;

# ANSI sequences for colours					# ANSI sequences for colours
my %colour;							my %colour;
$colour{white}       = "\033[1;37m";				$colour{white}       = "\033[1;37m";
$colour{yellow}      = "\033[1;33m";				$colour{yellow}      = "\033[1;33m";
$colour{green}       = "\033[1;32m";				$colour{green}       = "\033[1;32m";
$colour{blue}        = "\033[1;34m";				$colour{blue}        = "\033[1;34m";
$colour{cyan}        = "\033[1;36m";				$colour{cyan}        = "\033[1;36m";
$colour{red}         = "\033[1;31m";				$colour{red}         = "\033[1;31m";
$colour{magenta}     = "\033[1;35m";				$colour{magenta}     = "\033[1;35m";
$colour{black}       = "\033[1;30m";				$colour{black}       = "\033[1;30m";
$colour{darkwhite}   = "\033[0;37m";				$colour{darkwhite}   = "\033[0;37m";
$colour{darkyellow}  = "\033[0;33m";				$colour{darkyellow}  = "\033[0;33m";
$colour{darkgreen}   = "\033[0;32m";				$colour{darkgreen}   = "\033[0;32m";
$colour{darkblue}    = "\033[0;34m";				$colour{darkblue}    = "\033[0;34m";
$colour{darkcyan}    = "\033[0;36m";				$colour{darkcyan}    = "\033[0;36m";
$colour{darkred}     = "\033[0;31m";				$colour{darkred}     = "\033[0;31m";
$colour{darkmagenta} = "\033[0;35m";				$colour{darkmagenta} = "\033[0;35m";
$colour{darkblack}   = "\033[0;30m";				$colour{darkblack}   = "\033[0;30m";
$colour{OFF}         = "\033[0;0m";				$colour{OFF}         = "\033[0;0m";

# Default colours if /etc/colordiffrc or ~/.colordiffrc do no	# Default colours if /etc/colordiffrc or ~/.colordiffrc do no
my $plain_text = $colour{white};				my $plain_text = $colour{white};
my $file_old   = $colour{red};					my $file_old   = $colour{red};
my $file_new   = $colour{blue};					my $file_new   = $colour{blue};
my $diff_stuff = $colour{magenta};				my $diff_stuff = $colour{magenta};
my $cvs_stuff  = $colour{green};				my $cvs_stuff  = $colour{green};

# Locations for personal and system-wide colour configuration	# Locations for personal and system-wide colour configuration
my $HOME   = $ENV{HOME};					my $HOME   = $ENV{HOME};
my $etcdir = '/etc';						my $etcdir = '/etc';

my ($setting, $value);						my ($setting, $value);
my @config_files = ("$etcdir/colordiffrc", "$HOME/.colordiffr	my @config_files = ("$etcdir/colordiffrc", "$HOME/.colordiffr
my $config_file;						my $config_file;

foreach $config_file (@config_files) {				foreach $config_file (@config_files) {
    if (open(COLORDIFFRC, "<$config_file")) {			    if (open(COLORDIFFRC, "<$config_file")) {
        while (<COLORDIFFRC>) {					        while (<COLORDIFFRC>) {
            chop;						            chop;
            next if (/^#/ || /^$/);				            next if (/^#/ || /^$/);
            s/\s+//g;						            s/\s+//g;
            ($setting, $value) = split ('=');			            ($setting, $value) = split ('=');
            if (!defined $value) {				            if (!defined $value) {
                print STDERR "Invalid configuration line ($_)	                print STDERR "Invalid configuration line ($_)
                next;						                next;
            }							            }
            if ($setting eq 'banner') {				            if ($setting eq 'banner') {
                if ($value eq 'no') {				                if ($value eq 'no') {
                    $show_banner = 0;				                    $show_banner = 0;
                }						                }
                next;						                next;
            }							            }
            $setting =~ tr/A-Z/a-z/;				            $setting =~ tr/A-Z/a-z/;
            $value =~ tr/A-Z/a-z/;				            $value =~ tr/A-Z/a-z/;
            if (!defined $colour{$value}) {			            if (!defined $colour{$value}) {
                print STDERR "Invalid colour specification ($	                print STDERR "Invalid colour specification ($
                next;						                next;
            }							            }
            if ($setting eq 'plain') {				            if ($setting eq 'plain') {
                $plain_text = $colour{$value};			                $plain_text = $colour{$value};
            }							            }
            elsif ($setting eq 'oldtext') {			            elsif ($setting eq 'oldtext') {
                $file_old = $colour{$value};			                $file_old = $colour{$value};
            }							            }
            elsif ($setting eq 'newtext') {			            elsif ($setting eq 'newtext') {
                $file_new = $colour{$value};			                $file_new = $colour{$value};
            }							            }
            elsif ($setting eq 'diffstuff') {			            elsif ($setting eq 'diffstuff') {
                $diff_stuff = $colour{$value};			                $diff_stuff = $colour{$value};
            }							            }
            elsif ($setting eq 'cvsstuff') {			            elsif ($setting eq 'cvsstuff') {
                $cvs_stuff = $colour{$value};			                $cvs_stuff = $colour{$value};
            }							            }
            else {						            else {
                print STDERR "Unknown option in $config_file:	                print STDERR "Unknown option in $config_file:
            }							            }
        }							        }
        close COLORDIFFRC;					        close COLORDIFFRC;
    }								    }
}								}

# colordiff specfic options here.  Need to pre-declare if usi	# colordiff specfic options here.  Need to pre-declare if usi

# -----------------------------------------------------------	# -----------------------------------------------------------
# FIXME								# FIXME
#								#
# Remove this (contributed) code for now - the long options b	# Remove this (contributed) code for now - the long options b
# of extra diff options supplied on the command line.  On bal	# of extra diff options supplied on the command line.  On bal
# diff options is probably more important than having colordi	# diff options is probably more important than having colordi
# on the command line.						# on the command line.
#								#
# Need to find a proper fix for this so that this can be rein	# Need to find a proper fix for this so that this can be rein

###  GetOptions(						###  GetOptions(
###      "no-banner" => sub { $show_banner = 0 },		###      "no-banner" => sub { $show_banner = 0 },
###      "plain-text=s" => \&set_color,				###      "plain-text=s" => \&set_color,
###      "file-old=s"   => \&set_color,				###      "file-old=s"   => \&set_color,
###      "file-new=s"   => \&set_color,				###      "file-new=s"   => \&set_color,
###      "cvs-stuff=s"  => \&set_color,				###      "cvs-stuff=s"  => \&set_color,
###      "diff-stuff=s" => \&set_color				###      "diff-stuff=s" => \&set_color
###  );								###  );
# -----------------------------------------------------------	# -----------------------------------------------------------

if ($show_banner == 1) {					if ($show_banner == 1) {
    print STDERR "$app_name $version ($app_www)\n";		    print STDERR "$app_name $version ($app_www)\n";
    print STDERR "$copyright $author, $author_email\n\n";	    print STDERR "$copyright $author, $author_email\n\n";
}								}

my @inputstream;						my @inputstream;

if (defined $ARGV[0]) {						if (defined $ARGV[0]) {
    # More reliable way of pulling in arguments			    # More reliable way of pulling in arguments
    open2(\*INPUTSTREAM, undef, "diff", @ARGV);			    open2(\*INPUTSTREAM, undef, "diff", @ARGV);
    @inputstream = <INPUTSTREAM>;				    @inputstream = <INPUTSTREAM>;
    close INPUTSTREAM;						    close INPUTSTREAM;
}								}
else {								else {
    @inputstream = <STDIN>;					    @inputstream = <STDIN>;
}								}

# Input stream has been read - need to examine it 		# Input stream has been read - need to examine it 
# to determine type of diff we have.				# to determine type of diff we have.
#								#
# This may not be perfect - should identify most reasonably	# This may not be perfect - should identify most reasonably
# formatted diffs and patches					# formatted diffs and patches

my $diff_type = 'unknown';					my $diff_type = 'unknown';
my $record;							my $record;

DIFF_TYPE: foreach $record (@inputstream) {			DIFF_TYPE: foreach $record (@inputstream) {
    # Unified diffs are the only flavour having '+++' or '---	    # Unified diffs are the only flavour having '+++' or '---
    # at the start of a line					    # at the start of a line
    if ($record =~ /^(\+\+\+|---)/) {				    if ($record =~ /^(\+\+\+|---)/) {
        $diff_type = 'diffu';					        $diff_type = 'diffu';
        last DIFF_TYPE;						        last DIFF_TYPE;
    }								    }
    # Context diffs are the only flavour having '***'		    # Context diffs are the only flavour having '***'
    # at the start of a line					    # at the start of a line
    elsif ($record =~ /^\*\*\*/) {				    elsif ($record =~ /^\*\*\*/) {
        $diff_type = 'diffc';					        $diff_type = 'diffc';
        last DIFF_TYPE;						        last DIFF_TYPE;
    }								    }
    # Plain diffs have NcN, NdN and NaN etc.			    # Plain diffs have NcN, NdN and NaN etc.
    elsif ($record =~ /^[0-9,]+[acd][0-9,]+$/) {		    elsif ($record =~ /^[0-9,]+[acd][0-9,]+$/) {
        $diff_type = 'diff';					        $diff_type = 'diff';
        last DIFF_TYPE;						        last DIFF_TYPE;
    }								    }
							      >	    elsif ($record =~ /(\s\|\s|\s<$|\s>\s)/) {
							      >	        $diff_type = 'diffy';
							      >	        last DIFF_TYPE;
							      >	    }
}								}

my $inside_file_old = 1;					my $inside_file_old = 1;

foreach (@inputstream) {					foreach (@inputstream) {
    if ($diff_type eq 'diff') {					    if ($diff_type eq 'diff') {
        if (/^</) {						        if (/^</) {
            print "$file_old";					            print "$file_old";
        }							        }
        elsif (/^>/) {						        elsif (/^>/) {
            print "$file_new";					            print "$file_new";
        }							        }
        elsif (/^[1-9]/) {					        elsif (/^[1-9]/) {
            print "$diff_stuff";				            print "$diff_stuff";
        }							        }
        elsif (/^(Index: |={5,}|RCS file: |retrieving |diff )	        elsif (/^(Index: |={5,}|RCS file: |retrieving |diff )
            print "$cvs_stuff";					            print "$cvs_stuff";
        }							        }
        else {							        else {
            print "$plain_text";				            print "$plain_text";
        }							        }
    }								    }
    elsif ($diff_type eq 'diffc') {				    elsif ($diff_type eq 'diffc') {
        if (/^- /) {						        if (/^- /) {
            print "$file_old";					            print "$file_old";
        }							        }
        elsif (/^\+ /) {					        elsif (/^\+ /) {
            print "$file_new";					            print "$file_new";
        }							        }
        elsif (/^\*{4,}/) {					        elsif (/^\*{4,}/) {
            print "$diff_stuff";				            print "$diff_stuff";
        }							        }
        elsif (/^\*\*\* [0-9]+,[0-9]+/) {			        elsif (/^\*\*\* [0-9]+,[0-9]+/) {
            print "$diff_stuff";				            print "$diff_stuff";
            $inside_file_old = 1;				            $inside_file_old = 1;
        }							        }
        elsif (/^\*\*\* /) {					        elsif (/^\*\*\* /) {
            print "$file_old";					            print "$file_old";
        }							        }
        elsif (/^--- [0-9]+,[0-9]+/) {				        elsif (/^--- [0-9]+,[0-9]+/) {
            print "$diff_stuff";				            print "$diff_stuff";
            $inside_file_old = 0;				            $inside_file_old = 0;
        }							        }
        elsif (/^--- /) {					        elsif (/^--- /) {
            print "$file_new";					            print "$file_new";
        }							        }
        elsif (/^!/) {						        elsif (/^!/) {
            if ($inside_file_old == 1) {			            if ($inside_file_old == 1) {
                print "$file_old";				                print "$file_old";
            }							            }
            else {						            else {
                print "$file_new";				                print "$file_new";
            }							            }
        }							        }
        elsif (/^(Index: |={5,}|RCS file: |retrieving |diff )	        elsif (/^(Index: |={5,}|RCS file: |retrieving |diff )
            print "$cvs_stuff";					            print "$cvs_stuff";
        }							        }
        else {							        else {
            print "$plain_text";				            print "$plain_text";
        }							        }
    }								    }
    elsif ($diff_type eq 'diffu') {				    elsif ($diff_type eq 'diffu') {
        if (/^-/) {						        if (/^-/) {
            print "$file_old";					            print "$file_old";
        }							        }
        elsif (/^\+/) {						        elsif (/^\+/) {
            print "$file_new";					            print "$file_new";
        }							        }
        elsif (/^\@/) {						        elsif (/^\@/) {
            print "$diff_stuff";				            print "$diff_stuff";
        }							        }
        elsif (/^(Index: |={5,}|RCS file: |retrieving |diff )	        elsif (/^(Index: |={5,}|RCS file: |retrieving |diff )
            print "$cvs_stuff";					            print "$cvs_stuff";
        }							        }
        else {							        else {
            print "$plain_text";				            print "$plain_text";
        }							        }
    }								    }
							      >	    elsif ($diff_type eq 'diffy') {
							      >	        if (/\s<$/) {
							      >	            print "$file_old";
							      >	        }
							      >	        elsif (/\s\|\s/) {
							      >	            print "$diff_stuff";
							      >	        }
							      >	        elsif (/\s>\s/) {
							      >	            print "$file_new";
							      >	        }
							      >	        else {
							      >	            print "$plain_text";
							      >	        }
							      >	    }
    s/$/$colour{OFF}/;						    s/$/$colour{OFF}/;
    print "$_";							    print "$_";
}								}

sub set_color {						      |	#   FIXME - See earlier commented-out code
    my ($type, $color) = @_;				      |	### sub set_color {
							      |	###     my ($type, $color) = @_;
    $type =~ s/-/_/;					      |	### 
    eval "\$$type = \$colour{$color}";			      |	###     $type =~ s/-/_/;
}							      |	###     eval "\$$type = \$colour{$color}";
							      >	### }
Index: colordiff.xml
===================================================================
RCS file: /cvsroot/colordiff/colordiff/src/Attic/colordiff.xml,v
retrieving revision 1.1.2.9
diff --side-by-side -r1.1.2.9 colordiff.xml
<?xml version="1.0" encoding="ISO-8859-1"?>			<?xml version="1.0" encoding="ISO-8859-1"?>
<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//	<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.1.2//
 "/usr/share/sgml/docbook/dtd/xml/4.1.2/docbookx.dtd" [		 "/usr/share/sgml/docbook/dtd/xml/4.1.2/docbookx.dtd" [
 <!ENTITY diff "<command>diff</command>">			 <!ENTITY diff "<command>diff</command>">
 <!ENTITY colordiff "<application>colordiff</application>">	 <!ENTITY colordiff "<application>colordiff</application>">
]>								]>

<!-- mostly borrowed from bogofilter's bogofilter.xml file, o	<!-- mostly borrowed from bogofilter's bogofilter.xml file, o
     by eric raymond -->					     by eric raymond -->

<refentry id='colordiff.1'>					<refentry id='colordiff.1'>
  <refmeta>							  <refmeta>
    <refentrytitle>colordiff</refentrytitle>			    <refentrytitle>colordiff</refentrytitle>
    <manvolnum>1</manvolnum>					    <manvolnum>1</manvolnum>
  </refmeta>							  </refmeta>

  <refnamediv id='name'>					  <refnamediv id='name'>
    <refname>colordiff</refname>				    <refname>colordiff</refname>
    <refpurpose>a tool to colorize &diff; output</refpurpose>	    <refpurpose>a tool to colorize &diff; output</refpurpose>
  </refnamediv>							  </refnamediv>

<refsynopsisdiv id='synopsis'>					<refsynopsisdiv id='synopsis'>
  <cmdsynopsis>							  <cmdsynopsis>
    <command>colordiff</command>				    <command>colordiff</command>
    <arg><replaceable>&diff; options</replaceable></arg>	    <arg><replaceable>&diff; options</replaceable></arg>
    <arg choice='req'><replaceable>file1</replaceable></arg>	    <arg choice='req'><replaceable>file1</replaceable></arg>
    <arg choice='req'><replaceable>file2</replaceable></arg>	    <arg choice='req'><replaceable>file2</replaceable></arg>
  </cmdsynopsis>						  </cmdsynopsis>
</refsynopsisdiv>						</refsynopsisdiv>

<refsect1 id='description'><title>Description</title>		<refsect1 id='description'><title>Description</title>

<para>&colordiff; is a wrapper for &diff; and produces the sa	<para>&colordiff; is a wrapper for &diff; and produces the sa
&diff; but with coloured syntax highlighting at the command l	&diff; but with coloured syntax highlighting at the command l
readability. The output is similar to how a &diff;-generated 	readability. The output is similar to how a &diff;-generated 
in <application>Vim</application> or <application>Emacs</appl	in <application>Vim</application> or <application>Emacs</appl
appropriate syntax highlighting options enabled. The colour s	appropriate syntax highlighting options enabled. The colour s
read from a central configuration file or from a local user	read from a central configuration file or from a local user
<filename>~/.colordiffrc</filename> file.</para>		<filename>~/.colordiffrc</filename> file.</para>

<para>&colordiff; makes use of ANSI colours and as such will 	<para>&colordiff; makes use of ANSI colours and as such will 
ANSI colours can be used - typical examples are xterms and Et	ANSI colours can be used - typical examples are xterms and Et
console sessions.</para>					console sessions.</para>

<para>&colordiff; has been tested on various flavours of Linu	<para>&colordiff; has been tested on various flavours of Linu
OpenBSD, but should be broadly portable to other systems.</pa	OpenBSD, but should be broadly portable to other systems.</pa

</refsect1>							</refsect1>

<refsect1 id="usage"><title>Usage</title>			<refsect1 id="usage"><title>Usage</title>

<para>Use &colordiff; wherever you would normally use &diff;,	<para>Use &colordiff; wherever you would normally use &diff;,
to &colordiff;:</para>						to &colordiff;:</para>

<para>For example:						<para>For example:

<screen>							<screen>
$ colordiff file1 file2						$ colordiff file1 file2
$ diff -u file1 file2 | colordiff				$ diff -u file1 file2 | colordiff
</screen>							</screen>

</para>								</para>

<para>You can pipe the output to 'less', using the '-R' optio	<para>You can pipe the output to 'less', using the '-R' optio
terminal types may get better results using '-r' instead), wh	terminal types may get better results using '-r' instead), wh
the colour escape sequences, otherwise displayed incorrectly 	the colour escape sequences, otherwise displayed incorrectly 
'less':								'less':

<screen>							<screen>
$ diff -u file1 file2 | colordiff | less -R			$ diff -u file1 file2 | colordiff | less -R
</screen>							</screen>

</para>								</para>

<para>You may find it useful to make &diff; automatically cal	<para>You may find it useful to make &diff; automatically cal
<command>colordiff</command>. Add the following line to		<command>colordiff</command>. Add the following line to
<filename>~/.bashrc</filename> (or equivalent):			<filename>~/.bashrc</filename> (or equivalent):

<screen>							<screen>
alias diff=colordiff						alias diff=colordiff
</screen>							</screen>

</para>								</para>

<para>Any options passed to &colordiff; are passed through to	<para>Any options passed to &colordiff; are passed through to

<para>Alternatively, a construct such as 'cvs diff SOMETHING 	<para>Alternatively, a construct such as 'cvs diff SOMETHING 
included in <filename>~/.bashrc</filename> as follows:		included in <filename>~/.bashrc</filename> as follows:

<screen>							<screen>
function cvsdiff () { cvs diff $@ | colordiff; }		function cvsdiff () { cvs diff $@ | colordiff; }
</screen>							</screen>

</para>								</para>

<para>Or, combining the idea above using 'less':		<para>Or, combining the idea above using 'less':

<screen>							<screen>
function cvsdiff () { cvs diff $@ | colordiff |less -R; }	function cvsdiff () { cvs diff $@ | colordiff |less -R; }
</screen>							</screen>

</para>								</para>

<para>Note that the function name, cvsdiff, can be customized	<para>Note that the function name, cvsdiff, can be customized

</refsect1>							</refsect1>

<refsect1 id='files'><title>Files</title>			<refsect1 id='files'><title>Files</title>

<variablelist><varlistentry>					<variablelist><varlistentry>
  <term><filename>/etc/colordiffrc</filename></term>		  <term><filename>/etc/colordiffrc</filename></term>
  <listitem><para>Central configuration file. User-specific s	  <listitem><para>Central configuration file. User-specific s
  enabled by copying this file to <filename>~/.colordiffrc</f	  enabled by copying this file to <filename>~/.colordiffrc</f
  making the appropriate changes.</para></listitem>		  making the appropriate changes.</para></listitem>
</varlistentry></variablelist>					</varlistentry></variablelist>

</refsect1>							</refsect1>

<refsect1 id='bugs'><title>Bugs</title>				<refsect1 id='bugs'><title>Bugs</title>

<para>Bug reports and suggestions/patches to <email>davee@sun	<para>Bug reports and suggestions/patches to <email>davee@sun
please.</para>							please.</para>

<para>&colordiff; does not work for side-by-side diffs (<comm	<para>&colordiff; does not work for side-by-side diffs (<comm
-y</command>). Note that <emphasis>does not work</emphasis> s	-y</command>). Note that <emphasis>does not work</emphasis> s
the resulting output will not be colour-highlighted -- correc	the resulting output will not be colour-highlighted -- correc
still be produced.</para>					still be produced.</para>

</refsect1>							</refsect1>

<refsect1 id='author'><title>Author</title>			<refsect1 id='author'><title>Author</title>

<para>&colordiff; is written and maintained by Dave Ewart	<para>&colordiff; is written and maintained by Dave Ewart
<email>davee@sungate.co.uk</email>. This manual page and the	<email>davee@sungate.co.uk</email>. This manual page and the
source XML was written by Graham Wilson <email>bob@decoy.wox. |	source XML was written by Graham Wilson <email>graham@mknod.o
Debian and is maintained by the author.  Graham Wilson also m	Debian and is maintained by the author.  Graham Wilson also m
package for &colordiff;.</para>					package for &colordiff;.</para>

</refsect1>							</refsect1>

</refentry>							</refentry>
<!-- vim:set ts=2 sw=2 et: -->					<!-- vim:set ts=2 sw=2 et: -->
